File: project/src/main/java/com/expedia/ui/javademo/interfaces/RecyclerViewContract.java
package com.expedia.ui.javademo.interfaces;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.expedia.ui.javademo.ItemVH;

/**
 * Created by nbirla on 15/11/17.
 */

public interface RecyclerViewContract {
    ItemVH createHolder(LayoutInflater inflater, ViewGroup parent, int type);
    int getViewType(String type);
}

File: project/src/main/java/com/expedia/ui/recyclerview/interfaces/RecyclerViewContract.kt
package com.expedia.ui.recyclerview.interfaces

import android.view.ViewGroup
import android.view.LayoutInflater
import com.expedia.ui.recyclerview.ItemVH



/**
 * Created by nbirla on 15/11/17.
 */
interface RecyclerViewContract {
    fun createHolder(inflater: LayoutInflater, parent: ViewGroup, type: Int): ItemVH<*>
    fun getViewType(type: String): Int
}